src/gdsync.cpp

```diff
@@ -1894,6 +1903,8 @@ struct gds_qp *gds_create_qp(struct ibv_pd *pd, struct ibv_context *context,
         gqp->recv_cq.cq = qp->recv_cq;
         gqp->recv_cq.curr_offset = 0;
 
+        memcpy(qp_attr, &exp_qp_attr, sizeof(gds_qp_init_attr_t));
```

Comment:
why is this needed?

Reply:
The original algorithm sets some fields such as send_cq and recv_cq of qp_attr. Hence, the users can do the following:

gds_create_qp(..., &qp_attr, ...);
qp_attr.send_cq; // access send_cq here

This memcpy is for addressing this case.

Reply:
Just to confirm, we don't need that, right? those fields are accessible thru gds_qp

Reply:
I will remove the copy in the next revision. Note that the behavior of gds_create_qp() will be changed because qp_attr will change from an in/out to in-only param.